OpenRA.Game/Traits/RevealsShroud.cs

```diff
@@ -30,12 +34,17 @@ public RevealsShroud(RevealsShroudInfo info)
 
 		public void Tick(Actor self)
 		{
-			if (cachedLocation != self.Location)
+			if (shroudUpdateTick < info.VisibilityUpdateDelay)
+				shroudUpdateTick++;
```

Comment:
Add a `return` inside this `if` and remove `shroudUpdateTick == info.VisibilityUpdateDelay &&` from the `if` below.